### DISCUSSION TITLE
git:from-image with custom Procfile location

### DISCUSSION THREAD

**Discussion by manuelvanrijn** [dokku/dokku#5420] [Q&A] [answered]
I'm trying to use a docker image for 2 apps, each having their own Procfile within the docker image

```
/usr/src/app/Procfile.1
/usr/src/app/Procfile.2
```

I've specified the Procfile as described:

```
dokku ps:set first-app procfile-path /usr/src/app/Procfile.1
# I also tried: dokku ps:set first-app procfile-path Procfile.1
```

No when I run the deployment using `git:from-image` it keeps saying `---> No Procfile found in app image`

How do I have to specify the `Procfile` for a `git:from-image`?

**Comment by josegonzalez** [comment, marked as answer]
EDIT: As of 0.29.x, Dokku will extract the `Procfile` from an image deployed via `git:from-image`. See #5480 for more details.
---

The `Procfile` is pulled from the repository source, not from the built image, so what you are doing isn't possible. If you need the same image with multiple Procfiles, consider using a multi-stage docker build, adding the correct Procfile to the WORKDIR in the stage, tagging individual stages as their own images, and then deploying each image (they will otherwise mostly share the same root image, so a deploy of each will be relatively quick).

**Reply by manuelvanrijn** [reply]
Thanks. It wasn't clear to me that this feature didn't work for docker images.
Your suggestion will work, so I'll go that path.

**Reply by josegonzalez** [reply]
@manuelvanrijn this will change in 0.29.x as updated in my original post.
